experimental/yarpl/include/yarpl/Flowables.h

```diff
@@ -108,6 +108,25 @@ class Flowables {
     return Flowable<T>::create(std::move(lambda));
   }
 
+  template <typename T, typename TGenerator>
+  static Reference<Flowable<T>> fromGenerator(TGenerator generator) {
+    auto lambda = [generator = std::move(generator)]
+    (Subscriber<T>& subscriber, int64_t requested) {
+      int64_t generated = 0;
+      try {
+        while (generated < requested) {
+          subscriber.onNext(generator());
+          ++generated;
+        }
+        return std::make_tuple(static_cast<int64_t>(generated), false);
```

Comment:
It seems like the wrapper would result in the exact same code after optimization (it would be templated on the lambda). This API makes it easy to fuck up by either returning the wrong number (off-by-one seems easy) or `false` instead of `true` etc.